src/SocialAuthUserManager.php

```diff
@@ -272,6 +311,78 @@ public function loadUserByProperty($field, $value) {
   }
 
   /**
+   * Checks if user exist in entity.
+   *
+   * @param string $social_media_id
+   *   User's name on Provider.
+   * @param string $type
+   *   User's email address.
+   *
+   * @return false if user doesn't exist
+   *   Else return Drupal User Id associate with the account.
+   */
+
+  public function checkIfUserExists($type,$social_media_id) {
+    $storage = \Drupal::entityManager()->getStorage('social_auth');
+    // Perform query on social auth entity.
+    $query = \Drupal::entityQuery('social_auth');
+
+    // Check If user exist by using type and social_media_id.
+    $social_auth_user = $query->condition('type', $type)
+      ->condition('social_media_id', $social_media_id)
+      ->execute();
+
+    $user_data = $storage->load(reset($social_auth_user));
+
+    if(!$social_auth_user) {
+      return FALSE;
+    }
+    else{
+      // Return User ID.
+      return $user_data->get('user_id')->getValue()[0]['value'];
+    }
+  }
+
+  /**
+   * Add user record in Social Auth Entity.
+   *
+   * @param integer $user_id
+   *   Drupal User ID.
+   * @param string $type
+   *   Type of social network.
+   * @param string $social_media_id
+   *   Unique Social ID returned by social network
+   *
+   * @return True
+   *   if User record was created or
+   *   False otherwise
+   */
+  public function addUserRecord($user_id,$type,$social_media_id) {
+    // Make sure we have everything we need.
+    if (!$user_id || !$type || !$social_media_id) {
+      $this->loggerFactory
+        ->get($this->getPluginId())
+        ->error('Failed to add user record in social_auth entiy. User_id: @user_id, social_network_identifier: @social_network_identifier, social_media_id: @social_media_id',
+          array('@user_id' => $user_id, '@social_network_identifier' => $type, '@social_media_id' => $social_media_id));
+      return FALSE;
+    }
+    else{
+      // Add user record.
+      $user_info= SocialAuth::create([
```

Comment:
Better if you use the entityTypeManager you already injected
Example
$values = [
    'key'  => 'value',
    'key2'  => 'value2',
];
$this->entityTypeManager()->getStorage('social_auth')->create($values);